trytond/trytond/transaction.py

```diff
@@ -279,12 +291,16 @@ def stop(self, commit=False):
     def set_context(self, context=None, **kwargs):
         if context is None:
             context = {}
-        manager = _AttributeManager(context=self.context)
+        old_context = self.context
         ctx = self.context.copy()
         ctx.update(context)
         if kwargs:
             ctx.update(kwargs)
         self.context = ImmutableDict(ctx)
+        if old_context == self.context:
```

Comment:
Pour moi c'était surtout `if not context and not kwargs` (et éventuellement, `if all(k in self.context and self.context[k] == v for k, v in list(kwargs.items()) + list((context or {}).items())`)